addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1484,10 +1485,17 @@ export class OdooEditor extends EventTarget {
         });
         if (!range) return;
         let { startContainer: start, startOffset, endContainer: end, endOffset } = range;
+        const startBlock = closestBlock(start);
+        const endBlock = closestBlock(end);
+        // Do not join blocks in the following cases:
+        // 1. start or end share a common ancestoral block with the range.
+        // 2. Selection spanning multiple TD's.
+        // 3. startBlock and endBlock are completely selected.
```

Comment:
:+1: 
Just small remarks:

```suggestion
        // Do not join blocks in the following cases:
        // 1. start and end share a common ancestor block with the range
        // 2. selection spans multiple TDs
        // 3. selection starts at beginning of startBlock and  ends at end of
        //    endBlock
```